x-pack/plugin/otel-data/src/main/resources/index-templates/metrics-service_destination.10m@template.yaml

```diff
@@ -0,0 +1,48 @@
+---
+version: ${xpack.oteldata.template.version}
+index_patterns: ["metrics-service_destination.10m.otel-*"]
+priority: 130
+data_stream:
+  hidden: true
+allow_auto_create: true
+_meta:
+  description: aggregated APM metrics template installed by x-pack
+  managed: true
+composed_of:
+  - metrics@tsdb-settings
+  - otel@mappings
+  - metrics-otel@mappings
+  - semconv-resource-to-ecs@mappings
+  - metrics@custom
+  - metrics-otel@custom
+ignore_missing_component_templates:
+  - metrics@custom
+  - metrics-otel@custom
+template:
+  settings:
+    index:
+      mode: time_series
+  mappings:
+    properties:
+      data_stream.type:
+        type: constant_keyword
+        value: metrics
+      metricset:
+        properties:
+          interval:
+            type: constant_keyword
+            value: 10m
+          name:
+            type: constant_keyword
+            value: service_destination
+    dynamic_templates:
+      - ecs_ip:
+          mapping:
+            type: ip
+          path_match: [ "ip", "*.ip", "*_ip" ]
+          match_mapping_type: string
+      - all_strings_to_keywords:
+          mapping:
+            ignore_above: 1024
+            type: keyword
+          match_mapping_type: string
```

Comment:
Should we do something with these? We repeat them everywhere.

If I remember correctly, the idea was to implement these here in the template, so people can't overwrite these.

First idea would be to move them to `metrics-otel@mappings.yaml`, but that won't cover what we want.

Reply:
Yes, I think it would be better to have a separate component template for these. 

> If I remember correctly, the idea was to implement these here in the template, so people can't overwrite these.

I think it was the opposite. Users should have the ability to customize the dynamic mappings. Therefore, when extracting this to a component template, we need to import that template after the @custom component template.

Reply:
Yes, so what I mean: users should be able to change dynamic templates currently defined in `metrics-otel@mappings.yaml`, but not the ones above (`ecs_ip` and `all_strings_to_keywords`), because changing or removing those could easily break the data stream since it effects dimensions. Right?

So I'd move these two dynamic templates into a separate  components (maybe `metrics-otel-fixed@mappings`)  and metrics template would be something like:

```
composed_of:
  - metrics@tsdb-settings
  - otel@mappings
  - metrics-otel@mappings
  - semconv-resource-to-ecs@mappings
  - metrics@custom
  - metrics-otel@custom
  - metrics-otel-fixed@mappings #<- the new one
 ```
 
 (plus adding `metrics-[x]m.otel@custom` according to https://github.com/elastic/elasticsearch/pull/114042#discussion_r1787408686)

Does that sound ok?  

* Happy to take suggestion on the naming `metrics-otel-fixed@mappings` :)